hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -118,7 +118,6 @@ func (r *NodePoolReconciler) SetupWithManager(mgr ctrl.Manager) error {
 }
 
```

Comment:
>in a hotloop while trying to apply a `nil` value to the machinedeployment annotations field

fwiw we do use createOrUpdate for the resources the NodePool owns, e.g MachineDeployment. We only use patch helper for the NodePool itself.